### PR TITLE
Have build script add dependent modules (PSSA & Plaster)

### PR DESCRIPTION
**Pull request by tylerl0706**
This will add PSSA and Plaster to our build so that the vsix AppVeyor spits out _should_ be "release quality"

This vsix is a great example:
https://ci.appveyor.com/project/PowerShell/vscode-powershell/build/artifacts

### CODE REVIEW

**Comment by rkeithhill** [comment]
How about the `PowerShellEditorServices.Commands`?  Does it get copied elsewhere?

**Comment by tylerl0706** [comment]
I don't have that in my 1.6.0 modules folder? Where did you get that from?

**Comment by rkeithhill** [comment]
OK that is weird.  It is another module inside the `PowerShellEditorServices` folder.

**Comment by tylerl0706** [comment]
@rkeithhill yeah just checked. That's inside the PowerShellEditorServices folder so it will be included in the vsix that this change generates 😄

**Comment by rjmholt** [comment]
Looks like Save-Module has [minimum and maximum version parameters](https://docs.microsoft.com/en-us/powershell/module/powershellget/save-module?view=powershell-6), which is perfect -- so I'm thinking here would be the right place for us to import some json config file and have a project-wide module version specification.

This looks good to go to me!

**Comment by tylerl0706** [comment]
@rjmholt Added a modules.json file that includes all of our PowerShell module dependencies that we grab with Save-Module.

Note, since PowerShell get doesn't have a concept of exclusive maximum versions I just put a very high 1.X version to account for any breaking changes that might come in a 2.0.

**Comment by rjmholt** [comment]
Looks like `-AllowPrerelease` isn't supported by AppVeyor's version of `Save-Module`

**Comment by tylerl0706** [comment]
Because their versions of PowerShell don't have latest version of PowerShellGet. :(

**Comment by rjmholt** [comment]
Maybe
```powershell
$params= @{
   <supportedParams> = <values>
}
if ((Get-Command "Save-Module").Parameters.ContainsKey("AllowPrerelease"))
{
    $params += @{ "AllowPrerelease" = $_.Value.AllowPrerelease }
}
```

File: appveyor.yml [hunk 1]
```diff
@@ -19,9 +19,10 @@ install:
       Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force | Out-Null
       Install-Module InvokeBuild -RequiredVersion 3.2.1 -Scope CurrentUser -Force | Out-Null
       Install-Module platyPS -RequiredVersion 0.7.6 -Scope CurrentUser -Force | Out-Null
+  - powershell.exe -c "Install-Module PowerShellGet -Force"
 
 build_script:
-  - ps: Invoke-Build
+  - powershell.exe -c "Invoke-Build"
```
Comment (tylerl0706) [inline appveyor.yml]: have to do this because of updating PowerShellGet

**Comment by tylerl0706** [comment]
Ok, now the builds will update PowerShellGet and save the modules correctly. Can I get another sign off?

File: .travis.yml [hunk 2]
```diff
@@ -22,4 +22,5 @@ install:
 
 script:
   - ulimit -n 4096
-  - powershell -File build/travis.ps1
\ No newline at end of file
+  - sudo powershell -c "Install-Module PowerShellGet -Force"
+  - powershell -File build/travis.ps1
```
Comment (rkeithhill) [inline .travis.yml]: Does -File correctly report exit codes now?  At one time it didn't.

Comment (tylerl0706) [reply .travis.yml]: hmm not sure, I was just adding the install-module above it. Wasn't really focusing on the travis build script itself.

Comment (rjmholt) [reply .travis.yml]: Just tried `pwsh` in Windows 10:
```

Name                           Value
----                           -----
PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1-289-g18d0a0bcbd436b11611ec13f5b2b4b290e58b542
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

Writing `bad.ps1` as:
```powershell
throw "Bad!"
```
Gives:
```
PS1> pwsh -File .\bad.ps1
Bad!
At C:\Users\roholt\bad.ps1:1 char:1
+ throw "Bad!"
+ ~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (Bad!:String) [], RuntimeException
+ FullyQualifiedErrorId : Bad!

PS1> $LASTEXITCODE
1
```

Changing `bad.ps1` to:
```powershell
exit 1234
```
gives me:
```
PS1> pwsh -File .\bad.ps1

PS1> $LASTEXITCODE
1234
```

Comment (tylerl0706) [reply .travis.yml]: huh. Interesting.

Comment (rkeithhill) [reply .travis.yml]: Cool.  Looks like it got fixed somewhere along the way. I just tested in 5.1 and it works there too.  See Chris Oldwood's comment on this blog post - https://blogs.msdn.microsoft.com/powershell/2006/10/14/windows-powershell-exit-codes/

That is the behavior I remember.

Comment (tylerl0706) [reply .travis.yml]: Good to know! Thanks for that, @rkeithhill.

File: vscode-powershell.build.ps1 [hunk 3]
```diff
@@ -63,6 +65,20 @@ task Restore -If { "Restore" -in $BuildTask -or !(Test-Path "./node_modules") }
     exec { & npm install $logLevelParam }
 }
 
+task RestorePowerShellModules -If { -not (Test-Path "$PSScriptRoot/modules/Plaster") } {
+    $modules = Get-Content -Raw "./modules.json" | ConvertFrom-Json
```
Comment (SeeminglyScience) [inline vscode-powershell.build.ps1]: Quotes aren't needed here, but nbd.

File: appveyor.yml (hunk 1 above)
Comment (SeeminglyScience) [reply appveyor.yml]: Should we be more explicit here? (e.g. `-Command` instead of `-c`)

File: vscode-powershell.build.ps1 (hunk 3 above)
Comment (tylerl0706) [reply vscode-powershell.build.ps1]: added $PSScriptRoot

**Pull request merged**